inference-engine/tests/ngraph_functions/include/ngraph_functions/builders.hpp

```diff
@@ -22,8 +22,48 @@ ngraph::ParameterVector makeParams(const element::Type &type, const std::vector<
 ngraph::ParameterVector
 makeParams(const element::Type &type, const std::vector<std::pair<std::string, std::vector<size_t>>> &inputs);
 
-std::shared_ptr<ngraph::Node> makeConstant(const element::Type &type, const std::vector<size_t> &shape,
-                                           const std::vector<float> &data, bool random = false);
+template<typename T>
```

Comment:
Please move it to the `cpp` file

Reply:
It is template now and should be implemented in header, otherwise it could not be compiled for required type.

Reply:
In that case I believe you just need explicitly specialize the template for required types.
 
And why do you need templated `makeConstant` in bound of your PR? I think all indices that you generate for tests are easily fit float precision and will be successfully casted to necessary int type.

Reply:
In my opinion it's conveniently to use templates. There is enough routine to create tests. Do we really worry about test binary size to avoid templates here?